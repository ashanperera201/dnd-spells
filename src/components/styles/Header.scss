@import "../../styles/responsive";

.header-container {
    text-align: start;

    @include respond(phone) {
        text-align: center !important;
    }
}
