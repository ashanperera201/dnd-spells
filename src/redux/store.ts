import { createStore, combineReducers } from 'redux';
import favSpellsReducer from './fav-spells/fav-spell.reducer';

const reducers = combineReducers({
    spells: favSpellsReducer,
})

const store = createStore(
    reducers,
)

export default store;