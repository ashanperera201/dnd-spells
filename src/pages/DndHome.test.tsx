import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import DndHomePage from './DndHome';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { BrowserRouter, MemoryRouter } from 'react-router-dom'
import store from '../redux/store';
import { Provider as ReduxProvider } from "react-redux";

import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";


const theme = createTheme();

const homePageRender = (
    <React.StrictMode>
        <BrowserRouter>
            <ReduxProvider store={store}>
                <ThemeProvider theme={theme}>
                    <DndHomePage />
                </ThemeProvider>
            </ReduxProvider>
        </BrowserRouter>
    </React.StrictMode>
)

it('Should be renders dnd home page correctly.', () => {
    const tree = renderer.create(
        homePageRender
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Check whether the favorite button label is available or not.', () => {
    const { getAllByText } = render(
        homePageRender
    );
    expect(getAllByText('View Favorites')).toBeTruthy();
});

test('Check whether the button is available or not', () => {
    render(
        homePageRender
    );
    const button = screen.getByTestId('favorite-btn-element')
    userEvent.click(button);

    expect(screen.getAllByText("Favorite Spells")).toBeTruthy()
})