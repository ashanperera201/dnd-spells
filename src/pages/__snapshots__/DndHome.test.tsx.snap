// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be renders dnd home page correctly. 1`] = `
<div
  className="spell-container"
>
  <h3
    className="MuiTypography-root MuiTypography-h3 header-container css-1mocu2h-MuiTypography-root"
  >
    DND 5 Spells
  </h3>
  <div
    className="spell-container__action"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="favorite-btn-element"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      View Favorites
    </button>
  </div>
  <div
    className="spell-container__table"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1fw5qro-MuiPaper-root"
    >
      <div
        className="MuiTableContainer-root css-41abqd-MuiTableContainer-root"
      >
        <table
          aria-label="sticky table"
          className="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
                style={
                  Object {
                    "minWidth": 100,
                  }
                }
              >
                Spell Index
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
                style={
                  Object {
                    "minWidth": 80,
                  }
                }
              >
                Spell Name
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
                style={
                  Object {
                    "minWidth": undefined,
                  }
                }
              >
                Action
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            role={null}
          />
        </table>
      </div>
      <div
        className="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
      >
        <div
          className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            className="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            className="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
            id=":r1:"
          >
            Rows per page:
          </p>
          <div
            className="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
            onClick={[Function]}
            variant="standard"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby=":r1: :r0:"
              className="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
              id=":r0:"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              10
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              tabIndex={-1}
              value={10}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            className="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
          >
            0â€“0 of 0
          </p>
          <div
            className="MuiTablePagination-actions"
          >
            <button
              aria-label="Go to previous page"
              className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Go to previous page"
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next page"
              className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Go to next page"
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
